File: apps/ios/MarketRadar/MarketRadarApp/Models/MarketEvent.swift
import Foundation

enum EventType: String, Codable {
    case breakingNews = "breaking_news"
    case companyAnnouncement = "company_announcement"
    case earnings
    case guidance
    case macroEvent = "macro_event"
    case priceSpike = "price_spike"
    case volumeSpike = "volume_spike"
    case regulatoryFiling = "regulatory_filing"
    case dividend
    case shareBuyback = "share_buyback"
    case managementChange = "management_change"
}

enum EventSentiment: String, Codable {
    case positive
    case neutral
    case negative
}

struct MarketEvent: Identifiable, Codable {
    let id: String
    let eventType: EventType
    let title: String
    let summary: String
    let sourceName: String
    let sourceURL: String
    let publishedAt: String
    let receivedAt: String
    let affectedSymbols: [String]
    let importanceScore: Int
    let reliabilityScore: Int
    let sentiment: EventSentiment
    let confidence: Double
    let isMock: Bool

    enum CodingKeys: String, CodingKey {
        case id
        case eventType = "event_type"
        case title
        case summary
        case sourceName = "source_name"
        case sourceURL = "source_url"
        case publishedAt = "published_at"
        case receivedAt = "received_at"
        case affectedSymbols = "affected_symbols"
        case importanceScore = "importance_score"
        case reliabilityScore = "reliability_score"
        case sentiment
        case confidence
        case isMock = "is_mock"
    }
}

File: services/api/app/domain/market_event.py
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel, Field


class EventType(StrEnum):
    breaking_news = "breaking_news"
    company_announcement = "company_announcement"
    earnings = "earnings"
    guidance = "guidance"
    macro_event = "macro_event"
    price_spike = "price_spike"
    volume_spike = "volume_spike"
    regulatory_filing = "regulatory_filing"
    dividend = "dividend"
    share_buyback = "share_buyback"
    management_change = "management_change"


class Sentiment(StrEnum):
    positive = "positive"
    neutral = "neutral"
    negative = "negative"


class MarketEvent(BaseModel):
    id: str
    event_type: EventType
    title: str
    summary: str
    source_name: str
    source_url: str
    published_at: datetime
    received_at: datetime
    affected_symbols: list[str]
    importance_score: int = Field(ge=0, le=100)
    reliability_score: int = Field(ge=0, le=100)
    sentiment: Sentiment
    confidence: float = Field(ge=0, le=1)
    is_mock: bool
